catroid/src/main/java/org/catrobat/catroid/sensing/CollisionInformation.java

```diff
@@ -164,10 +164,9 @@ private void createCollisionPolygon(String path) {
 				ArrayList<PointF> points = getPointsFromPolygonVertices(boundingPolygon.get(i));
 				ArrayList<PointF> simplified = simplifyPolygon(points, 0, points.size() - 1, epsilon);
 
-				if (pointToPointDistance(simplified.get(0), simplified.get(simplified.size() - 1)) < epsilon) {
+				if (Math.hypot(simplified.get(0).x - simplified.get(simplified.size() - 1).x, simplified.get(0).y - simplified.get(simplified.size() - 1).y) < epsilon) {
```

Comment:
PRAISE: Actually, after seeing this solution, I am thinking your previous code for this was more readable and easily understandable. Sorry, my bad for suggesting the additional refactoring, I guess I didn't think enough about the complexity in the parameters. 

Could you please change it  back to your previous version, i.e., with the Math.hypot in pointToPointDistance? I guess also no explicit unit test is needed for the pointToPointDistance, though other reviewers might disagree. I apologize.

Thanks!

Reply:
Yes, I think this also, after this change it will take more time to be understood. I reverted now and add the pointToPointDistance method again. Thank you for reviewing ^_^